tests/unit/scheduler-test.js

```diff
@@ -136,4 +154,40 @@ module('Unit | Scheduler', function(hooks) {
 
     assert.equal(_getScheduleFn(), SIMPLE_CALLBACK);
   });
+
+  test('route idle and route painted schedule functions are correct', function(assert) {
+    assert.expect(2);
+
+    assert.equal(_getWhenRouteIdleScheduleFn(), requestIdleCallback);
```

Comment:
This seems somewhat brittle to me, I would prefer not to test the private APIs and rely on their coverage by way of the public APIs that we expose...

Reply:
Agreed, good call out — I've updated tests